### PR TITLE
Insertion sort

### PR DESCRIPTION
**Pull request by johnsonjew**

### CODE REVIEW

File: insertion_sort.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+from __future__ import unicode_literals
```
Comment (Jakeand3rson) [inline insertion_sort.py:1]: Good idea renaming it to be more specific.

File: insertion_sort.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+from time import time
+
+
+def insertion_sort(init_list):
```
Comment (Jakeand3rson) [inline insertion_sort.py:5]: Its a good idea to stay away from naming functions and files the same. It gets confusing real quick.

File: test_insertion.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+from __future__ import unicode_literals
+from insertion_sort import insertion_sort
+from random import shuffle
+
+
+def test_simple():
+    simple_list = [10, 7, 8, 9, 2, 15, 0, 5]
+    sorted_simple = insertion_sort(simple_list)
+    assert sorted_simple != simple_list
+    assert sorted_simple == sorted(simple_list)
```
Comment (Jakeand3rson) [inline test_insertion.py:8]: sorted(simple_list) is a cool way to test your sort with the built in. As long as you can guarantee the built in is working..

File: test_insertion.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    repeat_list = [10, 7, 8, 9, 2, 15, 0, 5, 15]
+    sorted_repeat = insertion_sort(repeat_list)
+    assert sorted_repeat != repeat_list
+    assert sorted_repeat == sorted(repeat_list)
+
+
+def test_already_sorted():
+    already_sorted = [0, 2, 5, 7, 8, 9, 10, 15]
+    sorted_already = insertion_sort(already_sorted)
+    assert sorted_already == already_sorted
```
Comment (Jakeand3rson) [inline test_insertion.py]: Kind of a weird test, but good idea!
Its a good edge case to test for.

File: test_insertion.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    sorted_reverse = insertion_sort(reverse_list)
+    assert sorted_reverse != reverse_list
+    assert sorted_reverse == sorted(reverse_list)
+
+
+def test_float():
+    float_list = [10, 7, 8, 9.5, 2.5, 15, 0, 5]
+    sorted_float = insertion_sort(float_list)
+    assert sorted_float != float_list
+    assert sorted_float == sorted(float_list)
```
Comment (Jakeand3rson) [inline test_insertion.py:32]: Another good edge case.

File: test_insertion.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+
+def test_long():
+    long_list = range(10000)
+    shuffle(long_list)
```
Comment (Jakeand3rson) [inline test_insertion.py]: `long_list = random.sample(range(10000), 10000)`
This is a way to do it in one line.
